src/Components/VideoListing/VideoListing/VideoCard.jsx

```diff
@@ -0,0 +1,52 @@
+import { Card } from "../../UI/";
+import { useState } from "react";
+import { VideoDropdown } from "./VideoDropdown";
+import "./VideoCard.css";
+export const VideoCard = ({
+  thumbnail,
+  glimpse,
+  title,
+  creator,
+  views,
+  date,
+  label,
+}) => {
+  const [cardHover, setCardHover] = useState(thumbnail);
+  const [showDropdown, setShowDropdown] = useState(false);
+
+  return (
+    <Card className="card-vertical card-video">
+      <div
+        className="card-top-right"
+        onMouseEnter={() => setShowDropdown(true)}
+        onMouseLeave={() => setShowDropdown(false)}
+      >
+        <i class="fa-solid fa-ellipsis-vertical"></i>
+        {showDropdown && <VideoDropdown />}
+      </div>
+      <div
+        className="card__img"
+        onMouseEnter={() => setCardHover(glimpse)}
+        onMouseLeave={() => setCardHover(thumbnail)}
```

Comment:
Great functionality here do check why it is broken the preview link.

Reply:
Yes I will fix asset related issue in next PR. Thank you for pointing it out.